src/reactors/V3DutchOrderReactor.sol

```diff
@@ -103,6 +109,23 @@ contract V3DutchOrderReactor is BaseReactor {
         }
     }
 
+    function _updateWithGasAdjustment(V3DutchOrder memory order) internal view {
+        // positive means an increase in gas
+        int256 gasDeltaGwei = block.basefee.sub(order.gasSnapshot);
+
+        // Gas increase should increase input
+        int256 inputDelta = int256(order.baseInput.adjustmentPerGweiBasefee) * gasDeltaGwei / 1 gwei;
```

Comment:
is gwei too large of a denominator? I wonder if wei or some percentage of a gwei would be better, then we can move this into a constant

Reply:
mulDivDown? or does that not support signed ints

Reply:
@marktoda yeah mulDivDown only supports uint256.

@zhongeric I was thinking about this a lot too.. I think gwei is the easiest to reason about with today's gas prices. The int256 gives us ~1e76 of space and if we assume gas is around 1e9-1e11 we get a range of 1 wei - 1e65 for possible `adjustmentPerGweiBasefee` values.

If we look at low value token like SHIB for example which has 18 decimals and is worth 0.0000000057064226 ETH, 1 gwei = 5.7064226 SHIB, so we'd want the `adjustmentPerGweiBasefee = 5706422600 * 10k gas` (ignoring tx call gas) and we'd still have 1e53 of wiggle room before overflowing.